### ISSUE TITLE
Unused internal methods of final classes are not reported as unused

### ISSUE THREAD

**Issue by parched** [microsoft/pyright#4014]
**Describe the bug**
Unused internal methods of final classes are not reported as unused.

**To Reproduce**
```
@final
class Something:
    def _my_func(self):
        pass
```

**Expected behavior**
I expected that `_my_func` would be reported as unused because it isn't used in this class or any subclass.

**Additional feature request**
(Please say if you'd like me to create a new issue for this)
It would be great if there was a pyright setting to change the leading underscore meaning for methods to "private" instead of "protected". We basically never use protected semantics, so it would be far more useful to get dead code warnings.

**VS Code extension or command-line**
Pylance language server 2022.9.40 (pyright 9fd1b94c)

**Comment by erictraut** [comment]
I'm not convinced that all methods starting with an underscore within a `@final` class should be considered unused. 

Consider this:

```python
class Parent:
    def _get_value(self) -> int:
        return 1

    def do_something(self):
        return self._get_value()

@final
class Child(Parent):
    def _get_value(self):
        """Override default value"""
        return 2
```

I understand your preferred behavior about "private" and "protected", but that's not consistent with the standards established in the Python community. The Python compiler treats class attributes and methods beginning with a double underscore as private and even helps to enforce this by doing name mangling.

**Comment by parched** [comment]
That's true. Although, in theory, that case could be detected, but probably not worth it.
